src/test/scala/com/cognite/sdk/scala/v1/TransformationsTest.scala

```diff
@@ -0,0 +1,208 @@
+// Copyright 2020 Cognite AS
+// SPDX-License-Identifier: Apache-2.0
+
+package com.cognite.sdk.scala.v1
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import com.cognite.sdk.scala.common._
+
+import java.time.temporal.ChronoUnit
+import java.time.Instant
+import scala.util.control.NonFatal
+import java.util.UUID
+
+@SuppressWarnings(
+  Array("org.wartremover.warts.NonUnitStatements")
+)
+class TransformationsTest extends CommonDataModelTestHelper with RetryWhile {
+
+  private val client = new GenericClient[IO](
+    "scala-sdk-test",
+    "extractor-bluefield-testing",
+    "https://bluefield.cognitedata.com",
+    authProvider,
+    None,
+    None,
+    None
+  )
+
+  def shortRandomUUID(): String = UUID.randomUUID().toString.substring(0, 8)
+
+  it should "list transformations" in {
+    val res = client.transformations.list(Some(5)).compile.toList.unsafeRunSync()
+    res.size shouldBe 5
+  }
+
+  it should "create and delete transformations" in {
+    val transformationsToCreate = (0 to 3).map { i =>
+      val uniquePrefix = shortRandomUUID()
+      TransformationCreate(
+        s"$uniquePrefix-transformation-sdk-test",
+        Some(s"select ${i.toString}"),
+        Some(GeneralDataSource("events")),
+        conflictMode = Some("upsert"),
+        Some(false),
+        sourceOidcCredentials = Some(credentials),
+        destinationOidcCredentials = Some(credentials),
+        externalId = s"$uniquePrefix-transformation-sdk-test",
+        ignoreNullFields = true,
+        dataSetId = None
+      )
+    }
+    val resCreates =
+      client.transformations.createItems(Items(transformationsToCreate)).unsafeRunSync()
+    resCreates.size shouldBe transformationsToCreate.size
+    resCreates.map(_.name) should contain theSameElementsAs transformationsToCreate.map(_.name)
+    resCreates.map(_.query) should contain theSameElementsAs transformationsToCreate.flatMap(
+      _.query
+    )
+    resCreates.map(_.destination) should contain theSameElementsAs transformationsToCreate.flatMap(
+      _.destination
+    )
+    resCreates.map(_.externalId) should contain theSameElementsAs transformationsToCreate.map(
+      _.externalId
+    )
+
+    val externalIds = transformationsToCreate.map(_.externalId)
+    retryWithExpectedResult[Seq[TransformationRead]](
+      {
+        client.transformations.delete(externalIds.map(CogniteExternalId(_)), true).unsafeRunSync()
+        client.transformations.retrieveByExternalIds(externalIds, true).unsafeRunSync()
+      },
+      t => t shouldBe empty
+    )
+  }
+
+  it should "retrieve transformations by ids or externalIds" in {
+    val transformationsToCreate = (0 to 3).map { i =>
+      val uniquePrefix = shortRandomUUID()
```

Comment:
Isn't the intention for this prefix to be the same for all the transformations we create in this test, ie. it should be above `val transformationsToCreate`, and we should use `s"$uniquePrefix-sdk-test-${i.toString}"` instead, as we do in the next test?

Reply:
yes I moved it outside and add `-${i.toString}` at the end